lib/rex/ui/text/dispatcher_shell.rb

```diff
@@ -422,12 +422,7 @@ def tab_complete_stub(str, quote: nil)
       end
     }
 
-    # Verify that our search string is a valid regex
-    begin
-      Regexp.compile(str,Regexp::IGNORECASE)
-    rescue RegexpError
-      str = Regexp.escape(str)
-    end
```

Comment:
@cgranleese-r7 What was this old code doing? Why is it no longer needed? 👀

Reply:
So Dean and I look at this and the code you're referring to was giving the unicode error due to not being escaped correctly before being compiled, we also looked into using `Regexp.new` but from what we understood it was basically the same as running `Rexexp.compile`. 

Unicode error:
```
warning: invalid Unicode Property \P: /C:\Pro/
```

So I had got to a point were I had this code:
```Ruby
 Regexp.new(Regexp.escape(utf8_buf), Regexp::IGNORECASE)
```

But when looking into it, the code above was not escaping the special characters. 

`Regexp.new(Regexp.escape(utf8_buf), Regexp::IGNORECASE)` output
```Ruby
/C:\\Pro/i
```

`str = Regexp.escape(str)` output
```Ruby
"C:\\\\Pro"
```

When testing it the previously code was returning the unicode error every-time, but if you called escape within the params of `Regexp.new` it didn't escape the string correctly, so we landed on this.

Reply:
@cgranleese-r7 That's the warning you were running into, but what was this code previously doing? Is it safe to delete? Are we losing functionality by deleting this? 👀

Reply:
For instance, try this tab completion against master: `use .*sharepoint.*xoml<tab>` and compare it to the results of your branch

Reply:
@adfoster-r7 Ah ok, yea that no longer allows tab completion on my branch. Ok, I'll take another look at this then and see what I can come up with.

Thanks for looking at that and giving me a lead on where to start, appreciate it 👍

Reply:
Threading the other conversation where we looked more into the problems with turning arbitrary text into regex:

https://github.com/rapid7/metasploit-framework/pull/15050#discussion_r616562443

Reply:
For the record, @adfoster-r7 and I discussed this implementation last year. You're safe to remove it. It's ancient, only a small handful of people know about it, and even fewer use it. No one will miss it, especially if it causes bugs.